akka-actor-tests/src/test/scala/akka/AkkaVersionSpec.scala

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka
+
+import org.scalatest.{Matchers, WordSpec}
+
+class AkkaVersionSpec extends WordSpec with Matchers {
+
+  "The Akka version check" must {
+
+    "succeed if version is ok" in {
+      AkkaVersion.require("AkkaVersionSpec", Set("2.5.6"), "2.5.6")
+      AkkaVersion.require("AkkaVersionSpec", Set("2.5.6"), "2.5.7")
+      AkkaVersion.require("AkkaVersionSpec", Set("2.5.6", "2.4.19"), "2.4.19")
+    }
+
+    "fail if minor version is missing" in {
+      intercept[UnsupportedAkkaVersion] {
+        AkkaVersion.require("AkkaVersionSpec", Set("2.5.6"), "2.4.19")
+      }
+    }
+
+    "fail if patch version is too low" in {
+      intercept[UnsupportedAkkaVersion] {
+        AkkaVersion.require("AkkaVersionSpec", Set("2.5.6"), "2.5.5")
+      }
+    }
+
+    "fail if Akka version is SNAPSHOT or timestamped snapshot" in {
+      intercept[UnsupportedAkkaVersion] {
+        AkkaVersion.require("AkkaVersionSpec", Set("2.5.6"), "2.5-SNAPSHOT")
+      }
+    }
+
+    "fail if required versions are invalid" in {
+      intercept[UnsupportedAkkaVersion] {
+        AkkaVersion.require("AkkaVersionSpec", Set("2.5-SNAPSHOT"), "2.5-SNAPSHOT")
```

Comment:
let's see, what does this mean? isn't this the case we will have if akka-http is built with 2.5-SNAPSHOT?

Reply:
Yeah, I'm not sure how to deal with the snapshots given that we do a sidestep from what I think is normal SNAPSHOT naming in maven land (that would be x.y.z-SNAPSHOT which is a snapshot of a upcoming x.y.z version)

Reply:
Shouldn't we just ignore everything that has to do with SNAPSHOT or timestamp snapshots? Then you are on your own.

Reply:
Sounds reasonable